sector-builder-ffi/src/reexported.rs

```diff
@@ -0,0 +1,305 @@
+    src_fd: libc::c_int,
+    src_size: u64,
+    dst_fd: libc::c_int,
+) -> *mut filecoin_proofs_ffi::types::WriteWithoutAlignmentResponse {
+    catch_panic_response(|| {
+        crate::api::init_log();
+
+        filecoin_proofs_ffi::api::write_without_alignment(src_fd, src_size, dst_fd)
+    })
+}
+
+/// TODO: document
+///
+#[no_mangle]
+pub unsafe extern "C" fn sector_builder_ffi_reexported_seal_pre_commit(
+    sector_class: filecoin_proofs_ffi::types::FFISectorClass,
+    cache_dir_path: *const libc::c_char,
+    staged_sector_path: *const libc::c_char,
+    sealed_sector_path: *const libc::c_char,
+    sector_id: u64,
+    prover_id: &[u8; 32],
+    ticket: &[u8; 32],
+    pieces_ptr: *const filecoin_proofs_ffi::types::FFIPublicPieceInfo,
+    pieces_len: libc::size_t,
+) -> *mut filecoin_proofs_ffi::types::SealPreCommitResponse {
+    catch_panic_response(|| {
+        crate::api::init_log();
+
+        filecoin_proofs_ffi::api::seal_pre_commit(
+            sector_class,
+            cache_dir_path,
+            staged_sector_path,
+            sealed_sector_path,
+            sector_id,
+            prover_id,
+            ticket,
+            pieces_ptr,
+            pieces_len,
+        )
+    })
+}
+
+/// TODO: document
+///
+#[no_mangle]
+pub unsafe extern "C" fn sector_builder_ffi_reexported_seal_commit(
+    sector_class: filecoin_proofs_ffi::types::FFISectorClass,
+    cache_dir_path: *const libc::c_char,
+    sector_id: u64,
+    prover_id: &[u8; 32],
+    ticket: &[u8; 32],
+    seed: &[u8; 32],
+    pieces_ptr: *const filecoin_proofs_ffi::types::FFIPublicPieceInfo,
+    pieces_len: libc::size_t,
+    spco: filecoin_proofs_ffi::types::FFISealPreCommitOutput,
+) -> *mut filecoin_proofs_ffi::types::SealCommitResponse {
+    catch_panic_response(|| {
+        crate::api::init_log();
+
+        filecoin_proofs_ffi::api::seal_commit(
+            sector_class,
+            cache_dir_path,
+            sector_id,
+            prover_id,
+            ticket,
+            seed,
+            pieces_ptr,
+            pieces_len,
+            spco,
+        )
+    })
+}
+
+/// TODO: document
+///
+#[no_mangle]
+pub unsafe extern "C" fn sector_builder_ffi_reexported_unseal(
+    sector_class: filecoin_proofs_ffi::types::FFISectorClass,
+    sealed_sector_path: *const libc::c_char,
+    unseal_output_path: *const libc::c_char,
+    sector_id: u64,
+    prover_id: &[u8; 32],
+    ticket: &[u8; 32],
+    comm_d: &[u8; 32],
+) -> *mut filecoin_proofs_ffi::types::UnsealResponse {
+    catch_panic_response(|| {
+        crate::api::init_log();
+
+        filecoin_proofs_ffi::api::unseal(
+            sector_class,
+            sealed_sector_path,
+            unseal_output_path,
+            sector_id,
+            prover_id,
+            ticket,
+            comm_d,
+        )
+    })
+}
+
+#[no_mangle]
+pub unsafe extern "C" fn sector_builder_ffi_reexported_destroy_verify_seal_response(
```

Comment:
why do you have to duplicate all of these? if this is parsed by cbindgen and they are reexported, they should show up

Reply:
@dignifiedquire - 

> they should show up

They do not show up, which is why I reexport them.

Reply:
@vmx could you take a look at this when you have a moment, this seems really unfortunate

Reply:
(independent of this merging)

Reply:
*lol* I saw this conversation in my emails and thought "I need to take a look at that" and read the comment that I should have a look afterwards :)

Reply:
@laser @dignifiedquire It's not a cbindgen issue, but a Rust one (https://github.com/rust-lang/rfcs/issues/2771). I don't know much about linking C libraries, but as far as I understand, the problem is that those C functions are not re-exported, hence cbindgen can't pick them up. The only workaround is to wrap them manually, which is what this PR is doing.

Reply:
Thanks for looking in to this, @vmx .